numpy/core/include/numpy/npy_3kcompat.h

```diff
@@ -99,32 +99,6 @@ static NPY_INLINE int PyInt_Check(PyObject *op) {
 
 #define Npy_EnterRecursiveCall(x) Py_EnterRecursiveCall(x)
 
-/* Py_SETREF was added in 3.5.2, and only if Py_LIMITED_API is absent */
-#if PY_VERSION_HEX < 0x03050200
-    #define Py_SETREF(op, op2)                      \
-        do {                                        \
-            PyObject *_py_tmp = (PyObject *)(op);   \
-            (op) = (op2);                           \
-            Py_DECREF(_py_tmp);                     \
-        } while (0)
-#endif
-
-/* introduced in https://github.com/python/cpython/commit/a24107b04c1277e3c1105f98aff5bfa3a98b33a0 */
-#if PY_VERSION_HEX < 0x030800A3
-    static NPY_INLINE PyObject *
-    _PyDict_GetItemStringWithError(PyObject *v, const char *key)
-    {
-        PyObject *kv, *rv;
-        kv = PyUnicode_FromString(key);
-        if (kv == NULL) {
-            return NULL;
-        }
-        rv = PyDict_GetItemWithError(v, kv);
-        Py_DECREF(kv);
-        return rv;
-    }
-#endif
```

Comment:
Was this Intentional?  But can just keep it...

```
numpy/core/src/umath/_rational_tests.c
```
and more importantly
```
numpy/core/src/common/npy_cpu_features.c
```
still have a few uses, would be nice to include at least the latter?

Reply:
This code block is only compatibility to support now unsupported (`< 3.8a3`) Python versions, we no longer need it, right?

I just noticed these when touching the file and decided to clean them up. Not attached to this change though.

Reply:
Yes, happy to keep it, just making a note in case it wasn't intentional.

Do you want to follow-up on those two files?  Otherwise, feel free to just merge yourself.

Reply:
Sorry, I'm not seeing what you mean in those files (I am sleep-deprived, so I could be dense here). This is deleting dead code, the functions themselves are now coming from Python and I didn't mean to touch how they are used.

Reply:
Either way, happy to get this in:)

Reply:
those files still seem to use `NPY_INLINE`

Reply:
Answer: those were regular source files that were added to `.gitignore` 🤦🏼. guess I'll write some code to check that whole long list. we can delete it all from `.gitignore` once we have fully switched to meson, just excluding `build/` is enough.